### PR TITLE
refactor: upgrade FOCA & major app rewrite

### LINKED ISSUE #153: Migrate boilerplate to FOCA v0.10.0
A lot of the code shared between this service and others in the organizationhas been moved to the [FOCA](https://github.com/elixir-cloud-aai/foca) archetype.

Duplicate code should be removed from this service and, where necessary, the remaining code should be refactored to make use of FOCA.

### PR DESCRIPTION
**Pull request by kushagra189**
**Details**

Excess boilerplate code was removed and replaced by FOCA's latest version.
Fixes #153

### CODE REVIEW

File: cwl_wes/config.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                    - v1.0
+        workflow_engine_versions:
+            cwl-tes: 0.2.0
+        default_workflow_engine_parameters:
+            - type: string
+              default_value: some_string
+            - type: int
+              default_value: '5'
+        tags:
+            known_tes_endpoints: 'https://tes.tsi.ebi.ac.uk/|https://tes-dev.tsi.ebi.ac.uk/|https://csc-tesk.c03.k8s-popup.csc.fi/|https://tesk.c01.k8s-popup.csc.fi/'
```
Comment (lvarin) [inline cwl_wes/config.yaml]: Just a detail, but the `c03.k8s-popup.csc.fi` cluster has been retired. Maybe not for this MR, but the examples could be fixed one of these days :)

**Comment by kushagra189** [comment]
This is currently blocked by #250 . #250 will be implemented separately and merged in this branch before releasing the refactored code.

**Comment by lvarin** [comment]
Can we merge this? Current default branch does not pass the tests, the image used in the docker file does not have any repository available.

**Review by uniqueg — APPROVED** [review]
Excellent work!

Of course, there's the `cwl-tes` import issue that should prevent the app from processing any workflows. But given that the `HEAD` commit is anyway broken and this PR is already gigantic, I think we can merge this now.

There are probably a bunch of other small issues as well, but overall it looks very good, I think. Again, time to merge and then figure out issues later on and get the app back working :rocket:

**Pull request merged**